### PR TITLE
When retrieving manga details from source, ensure title is pulled from source rather than local DB

### LINKED ISSUE #3608: [Bug] Title does not update from source along with other metadata.
### Device information
* Tachiyomi version: v0.10.3
* Android version: 8.0
* Device: Galaxy Tab S

## Steps to reproduce
1. Have a manga in your library.
2. Given source updates the title of the manga.
3. Refresh to update cover/metadata.

### Expected behavior
Title should update if changed to reflect the title used in the source.

### Actual behavior
Title remains the same until the manga is removed from library and cache is fully cleared.

### Other details
This is a fairly rare problem, but here's the real world case: The main title for "Ganbare, Douki-chan" was originally "Senpai is Mine" and MangaDex has updated their main display title to reflect this change but Tachiyomi won't update the title without fully removing the manga from the database and then doing a DB clean to remove not-in-library titles (wiping out any non-library progress in the process). It should probably update in line with the source with less friction to the end-user.

Not related to #2190

### PR DESCRIPTION
**Pull request by FlaminSarge**
Alongside a previous change, this ensures that manga titles for non-Library manga are always up to date, where previously an old/outdated title could be persisted in the DB and never updated until DB clear (#6177)

Closes #3608

### CODE REVIEW

**Comment by FlaminSarge** [comment]
I checked that an outdated title was updated properly when swipe-refreshing a manga details page when the manga was not in library, and that the new behavior did not occur if the manga was in library (to prevent the app from arbitrarily changing a user's library titles). In addition, every place that uses getMangaDetails subsequently uses manga.copyFrom(networkManga.toSManga()) to pull data from the resulting MangaInfo object, so this updated title behavior will only be picked up for cases that specifically update the manga title (alongside copyFrom) in the caller as well.

I have open questions about why these are set up the way they are, but that can be a discussion for later.

**Comment by arkon** [comment]
As others mentioned on Discord, you'd probably want to rename the downloads directory if present.

**Comment by quangkieu** [comment]
Hey I wonder if this is relative to kissmanga keep missing cover whenever initialized or favorite?

File: app/src/main/java/eu/kanade/tachiyomi/source/Source.kt [hunk 1]
```diff
@@ -67,6 +67,7 @@ interface Source : tachiyomi.source.Source {
         val sManga = manga.toSManga()
         val networkManga = fetchMangaDetails(sManga).awaitSingle()
         sManga.copyFrom(networkManga)
+        sManga.title = networkManga.title
```
Comment (FourTOne5) [inline app/src/main/java/eu/kanade/tachiyomi/source/Source.kt:70]: What not do it inside the `.copyFrom()` function?

Comment (FlaminSarge) [reply app/src/main/java/eu/kanade/tachiyomi/source/Source.kt:70]: I was under the impression that copyFrom was also used in places that wouldn't necessarily want to copy the title. My initial impl had a param on copyFrom for copying title that defaulted to false, and that version is probably also valid. If you feel strongly about it I can update to that instead.

**Comment by FlaminSarge** [comment]
Still pending changes for renaming downloaded folders, moving to Draft for now

File: app/src/main/java/eu/kanade/tachiyomi/source/Source.kt (hunk 1 above)
Comment (FourTOne5) [reply app/src/main/java/eu/kanade/tachiyomi/source/Source.kt:70]: Check if the network manga title is blank?

**Pull request closed without merging**